cmd/mdatagen/templates/metrics.go.tmpl

```diff
@@ -93,6 +134,38 @@ var MapAttribute{{ $name.Render }} = map[string]Attribute{{ $name.Render }}{
 {{ end }}
 {{- end }}
 
+{{ range $name, $info := .ResourceAttributes }}
+	{{- if $info.Enum -}}
+		// ResourceAttribute{{ $name.Render }} specifies the a value {{ $name }} resource attribute.
+		type ResourceAttribute{{ $name.Render }} int
+
+		const (
+		_ ResourceAttribute{{ $name.Render }} = iota
+		{{- range $info.Enum }}
+			ResourceAttribute{{ $name.Render }}{{ . | publicVar }}
+		{{- end }}
+		)
+
+		// String returns the string representation of the ResourceAttribute{{ $name.Render }}.
+		func (av ResourceAttribute{{ $name.Render }}) String() string {
+		switch av {
+		{{- range $info.Enum }}
+			case ResourceAttribute{{ $name.Render }}{{ . | publicVar }}:
+			return "{{ . }}"
+		{{- end }}
+		}
+		return ""
+		}
+
+		// MapResourceAttribute{{ $name.Render }} is a helper map of string to ResourceAttribute{{ $name.Render }} resource attribute value.
+		var MapResourceAttribute{{ $name.Render }} = map[string]ResourceAttribute{{ $name.Render }}{
+		{{- range $info.Enum }}
+			"{{ . }}": ResourceAttribute{{ $name.Render }}{{ . | publicVar }},
+		{{- end }}
+		}
+
+	{{ end }}
+{{- end }}
```

Comment:
I don't think we need this API. We just need to update `With<resource attribute>` options to read the settings and don't add the attribute if it's disabled